src/cli/compiler.js

```diff
@@ -51,21 +61,143 @@ class Compiler {
         : path.join(this.options.outputDir, 'subgraph.yaml')
 
     this.logger.info('')
-    this.logger.info(chalk.green('Completed'))
+    this.logger.info(chalk.green('Build completed'))
     this.logger.info('')
     this.logger.info('%s %s', chalk.bold(chalk.blue('Subgraph:')), hashOrFilename)
     this.logger.info('')
   }
 
   loadSubgraph() {
     try {
-      this.logger.step('Load subgraph:', this.options.subgraphManifest)
       return Subgraph.load(this.options.subgraphManifest)
     } catch (e) {
       this.logger.fatal('Failed to load subgraph:', e)
     }
   }
 
+  getFilesToWatch() {
+    try {
+      let allFiles = []
+      let subgraph = this.loadSubgraph()
+
+      // Add all file paths specified in manifest
+      allFiles.push(subgraph.getIn(['schema', 'file']))
+      subgraph.get('dataSources').map(dataSource => {
+          allFiles.push(dataSource.getIn(['mapping', 'file']))
+          dataSource.getIn(['mapping', 'abis']).map(abi => {
+            allFiles.push(abi.get('file'))
+          })
+      })
+
+      // Make paths absolute
+      let allAbsolutePaths = allFiles.map(file => {
+        return path.resolve(this.sourceDir, file)
+      })
+      return allAbsolutePaths
+    } catch(e) {
+      this.logger.fatal('Failed to parse subgraph file locations:', e)
+    }
+  }
+
+  watchAndCompile() {
+    let compiler = this
+    if (cluster.isMaster) {
```

Comment:
Is there any reason why we can't just catch exceptions thrown by `compile()` and keep watching? This cluster dance seems obscure but perhaps there are reasons for it?

Reply:
I went with try catch originally, updating the `fatal` logger to no longer exit the process, but was getting a cascade of errors as it moved through the try catch blocks out to `compile()`.    These errors make sense, but it's a bit cumbersome and makes it less 'falling of the log easy' to pinpoint the issue.

Reply:
Creating an exception constructor to throw the error and context message and then only logging at the top level.